apps/bluetooth/app.js

```diff
@@ -112,14 +113,18 @@ module.exports = function (activity) {
     logger.debug(`speak: ${text}`)
     if (!textIsEmpty(text)) {
       return activity.setForeground().then(() => {
-        if (wifi.getWifiState() === wifi.WIFI_CONNECTED) {
-          return activity.tts.speak(text, { impatient: false }).catch((err) => {
-            logger.error('play tts error: ', err)
-          })
-        } else if (alternativeVoice != null) {
-          logger.debug('No wifi connection, play alternative voice.')
-          return activity.playSound(alternativeVoice)
-        }
+        network.getStatus('NETWORK').then((reply) => {
+          var msg = JSON.parse(reply.msg[0])
```

Comment:
I think it's not user's duty to parse msg, it should be wrapped in network component.